tool/prelude.lua

```diff
@@ -32,6 +32,25 @@ function table.indexof(list, element)
   return -1
 end
 
+function table.checkifinbounds(seq, lowerbound, upperbound)
+  for i in ipairs(seq) do
```

Comment:
`ipairs` returns a tuple with 2 elements. The first one is the index, the second one the value. You probably want `for i, value in ipairs(seq) do` and then use `value` in the body.